demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -94,6 +92,197 @@ def _is_incident_type_default_playbook_found(self, incident_type_data):
+            layouts_container_file_path (str): Path to the file.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        layouts_container_id = list(layouts_container_data.keys())[0]
+        layouts_container = layouts_container_data.get(layouts_container_id, {})
+        layouts_container_name = layouts_container.get('name', layouts_container_id)
+        layouts_container_tabs = self._get_layouts_container_tabs(layouts_container)
+        scripts_set = set(get_layouts_scripts_ids(layouts_container_tabs))
+
+        # Check if the layouts container's scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_not_in_id_set)
+            error_message, error_code = Errors.layouts_container_non_existent_script_id(layouts_container_name,
+                                                                                        scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path=layouts_container_file_path):
+                is_valid = True
+
+        return is_valid
+
+    def _is_layout_scripts_found(self, layout_data, layout_file_path=None):
+        """Check if scripts of a layout  is in the id_set
+
+        Args:
+            layout_data (dict): Dictionary that holds the extracted details from the given layout.
+            layout_file_path (str): Path to the file.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        layout_id = list(layout_data.keys())[0]
+        layout = layout_data.get(layout_id, {})
+        layout_name = layout.get('typename', layout_id)
+        scripts = layout.get('scripts', [])
+        scripts_set = set(scripts)
+
+        # Check if the layouts container's scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_not_in_id_set)
+            error_message, error_code = Errors.layout_non_existent_script_id(layout_name,
+                                                                             scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path=layout_file_path):
+                is_valid = True
+
+        return is_valid
+
+    def _get_scripts_that_are_not_in_id_set(self, scripts_set):
```

Comment:
maybe use `scripts_in_entity` instead of `scripts_set` as the naming is confusing later on in the method